sw/airborne/modules/ctrl/optical_flow_landing.h

```diff
@@ -61,26 +69,41 @@ struct OpticalFlowLanding {
   float dgain_adaptive;         ///< D-gain for adaptive gain control
   int COV_METHOD;               ///< method to calculate the covariance: between thrust and div (0) or div and div past (1)
   int delay_steps;              ///< number of delay steps for div past
+  int window_size;              ///< number of time steps in "window" used for getting the covariance
+  float reduction_factor_elc;   ///< reduction factor - after oscillation, how much to reduce the gain...
+  float lp_cov_div_factor;      ///< low-pass factor for the covariance of divergence in order to trigger the second landing phase in the exponential strategy.
+  int count_transition;         ///< how many time steps the drone has to be oscillating in order to transition to the hover phase with reduced gain
+  float p_land_threshold;       ///< if during the exponential landing the gain reaches this value, the final landing procedure is triggered
 };
 
 extern struct OpticalFlowLanding of_landing_ctrl;
 
 // arrays containing histories for determining covariance
-float thrust_history[COV_WINDOW_SIZE];
-float divergence_history[COV_WINDOW_SIZE];
-float past_divergence_history[COV_WINDOW_SIZE];
+float thrust_history[MAX_COV_WINDOW_SIZE];
+float divergence_history[MAX_COV_WINDOW_SIZE];
+float past_divergence_history[MAX_COV_WINDOW_SIZE];
+float dt_history[MAX_COV_WINDOW_SIZE];
 unsigned long ind_hist;
 
 // Without optitrack set to: GUIDANCE_H_MODE_ATTITUDE
-// With optitrack set to: GUIDANCE_H_MODE_HOVER
-#define GUIDANCE_H_MODE_MODULE_SETTING GUIDANCE_H_MODE_HOVER
+// With optitrack set to: GUIDANCE_H_MODE_HOVER / NAV
+#define GUIDANCE_H_MODE_MODULE_SETTING GUIDANCE_H_MODE_NAV
 
 // Own guidance_v
 #define GUIDANCE_V_MODE_MODULE_SETTING GUIDANCE_V_MODE_MODULE
 
 // supporting functions for cov calculation:
```

Comment:
I based this on the example module. You're free to change it if it then still works.